ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerImpl.java

```diff
@@ -1561,7 +1561,8 @@ private InstallSnapshotReplyProto installSnapshotImpl(InstallSnapshotRequestProt
     }
 
     if (reply != null) {
-      if (request.hasLastRaftConfigurationLogEntryProto()) {
+      if (request.hasLastRaftConfigurationLogEntryProto() && reply.getResult() ==
+              InstallSnapshotResult.SNAPSHOT_INSTALLED) {
```

Comment:
SNAPSHOT_INSTALLED result is returned when SM is notified to install snapshot. For the case when snapshot is sent as part of installSnapshot request by Ratis,  SUCCESS is returned as the result.

Reply:
Oops, I missed that. Now I added the difference check to reduce the frequency of resetting conf. Coz this is redundant to overwrite the conf for each request.